src/main/java/org/sunbird/assessment/service/AssessmentServiceV2Impl.java

```diff
@@ -34,146 +39,329 @@ public class AssessmentServiceV2Impl implements AssessmentServiceV2 {
+                        for (Map<String, Object> section : sections) {
+                            questionsFromAssessment.addAll((List<String>) section.get(Constants.CHILD_NODES));
+                        }
+                        //Out of the list of questions received in the payload, retaining only those ids which are a part of the user's latest assessment
+                        identifierList.retainAll(questionsFromAssessment);
+                        //Fetching all the remaining questions details from the Redis
+                        List<Object> map = redisCacheMgr.mget(identifierList);
+                        for (int i = 0; i < map.size(); i++) {
+                            if (ObjectUtils.isEmpty(map.get(i))) {
+                                //Adding the not found keys as a seperate list
+                                newIdentifierList.add(identifierList.get(i));
+                            } else {
+                                //Filtering the details of the questions which are found in redis and adding them to another list
+                                questionList.add(filterQuestionMapDetail((Map<String, Object>) map.get(i)));
+                            }
+                        }
+                        //Taking the list which was formed with the not found values in Redis, we are making an internal POST call to Question List API to fetch the details
+                        if (!newIdentifierList.isEmpty()) {
+                            Map<String, Object> questionMapResponse = readQuestionDetails(newIdentifierList);
+                            if (questionMapResponse != null && !questionMapResponse.isEmpty() && Constants.OK.equalsIgnoreCase((String) questionMapResponse.get(Constants.RESPONSE_CODE))) {
+                                List<Map<String, Object>> questionMap = ((List<Map<String, Object>>) ((Map<String, Object>) questionMapResponse
+                                        .get(Constants.RESULT)).get(Constants.QUESTIONS));
+                                for (Map<String, Object> question : questionMap) {
+                                    if (!ObjectUtils.isEmpty(questionMap)) {
+                                        redisCacheMgr.putCache(Constants.QUESTION_ID + question.get(Constants.IDENTIFIER), question);
+                                        questionList.add(filterQuestionMapDetail(question));
+                                    } else {
+                                        logger.error(String.format("Failed to get Question Details for Id: %s", question.get(Constants.IDENTIFIER).toString()));
+                                    }
+                                }
+                            } else {
+                                logger.error(String.format("Failed to get Question Details from the Question List API for the IDs: %s", newIdentifierList.toString()));
+                            }
```

Comment:
We just log error ? what if this is the 1st time, there won't be any information in the cache... in that case, we simply log error and return empty successful response ?
we should through error even if we are not return one question details